shell.c

```diff
@@ -493,17 +497,86 @@ char* readLine() {
     }
 }
 
+// Display the result after processing and extraction
+void display_assignment_result(struct evaluation_factors e, char *line) {
+    bool res = is_print_command_valid(line);
+    if (!res) {
+        printf("%s\n", "Command is invalid!");
+        return;
+    }
+
+    int start_index = get_index_of_char_in_string(e.assignment);
+    char *variable_str = extract_substring_before_equal_sign(e.assignment);
+    char *value_str = extract_value_after_equal_sign(e.assignment, start_index);
+
+    char *display_str = retrieve_string_inside_quotation_mark(line);
+    // If '$' is not present (no assignment evaluation displayed)
+    if (is_dollar_present(display_str)) {
+        int pre_suf_index = get_index_of_character(display_str);
+
+        // Extract prefix (anything before $)
+        char *prefix = extract_prefix_from_string(display_str, pre_suf_index);
+        // Extract suffix (anything after the assignment of $)
+        char *suffix = extract_suffix_from_string(display_str, variable_str, pre_suf_index);
+
+        printf("%s%s%s\n", prefix, value_str, suffix);
+    } else {    // Display string as usual
+        printf("%s\n", display_str);
+    }
+}
+
+// Evaluate if the assignment is syntax-correct
+struct evaluation_factors verify_assignment_syntax(struct evaluation_factors e, char *line) {
+    /* Validate:
+        - Variable is followed by "="
+        - Only 1 "=" in the assignment
+    */
+    int index = 0;
+    int equal_sign_count = 0;
+    while (index < strlen(line)) {
+        if (line[index] == '=') equal_sign_count += 1;
+        index += 1;
+    }
+
+    if (equal_sign_count == 1) {
+        e.assignment = line;
+        e.valid = true;
+    }
+    return e;
+}
+
 void loop() {
     char *line;
     struct job *job;
 
-    while(1) {
+    // Initialize values for assignment evaluation preventing error prompt
+    struct evaluation_factors eval;
+    eval.assignment = "";
+    eval.valid = false;
+
+    while (1) {
         printf("> ");
         line = readLine();
+
         if(strlen(line) == 0) {
             checkZombie();
             continue;
         }
+
+        // Validate the assignment input
+        if (strlen(eval.assignment) == 0) {
+            eval = verify_assignment_syntax(eval, line);
+            continue;
+        }
+        
+        // Confirm that the assessment is valid
+        if (eval.valid == true) {
```

Comment:
doesn't it breaks job control functionality? I think shell substitution should be handled in job/process scope.

Reply:
Yes, for this part I still don't know the flow of your part thus I just have to put it here to test first.

Reply:
let's discuss this tomorrow in real life. For example executing `echo "Logged in user are $USERS"` should create a job with one process but in this case it will print substituted string.